pkg/network/netlink/socket.go

```diff
@@ -201,56 +201,34 @@ func (s *Socket) ReceiveAndDiscard() (bool, uint32, error) {
 // ReceiveInto reads one or more netlink.Messages off the socket
 func (s *Socket) ReceiveInto(b []byte) ([]netlink.Message, uint32, error) {
 	var netns uint32
-	var ret []netlink.Message
-	for {
-		n, oobn, err := s.recvmsg()
-		if err != nil {
-			return nil, 0, os.NewSyscallError("recvmsg", err)
-		}
+	n, oobn, err := s.recvmsg()
+	if err != nil {
+		return nil, 0, os.NewSyscallError("recvmsg", err)
+	}
 
-		n = nlmsgAlign(n)
-		// If we cannot fit the date into the supplied buffer,  we allocate a slice
-		// with enough capacity. This should happen very rarely.
-		if n > len(b) {
-			b = make([]byte, n)
-		}
-		copy(b, s.recvbuf[:n])
+	n = nlmsgAlign(n)
+	// If we cannot fit the date into the supplied buffer,  we allocate a slice
+	// with enough capacity. This should happen very rarely.
+	if n > len(b) {
+		b = make([]byte, n)
+	}
+	copy(b, s.recvbuf[:n])
+
+	msgs, err := ParseNetlinkMessage(b[:n])
+	if err != nil {
+		return nil, 0, err
+	}
 
-		msgs, err := ParseNetlinkMessage(b[:n])
+	if oobn > 0 && netns == 0 {
```

Comment:
why checking netns == 0 it's always true
seems old artifact

Reply:
Yes, redundant now without the `for` loop which I removed.